MT-7 Energy Magnum

ScarletLightweaverSS14 opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
Adding a new revolver to give head of secuirty, warden, detective, bso or replace captains laser gun for a cooler one

## Why we need to add this
Adding more variety to selection of energy weapons as it is very limited right now

## Media (Video/Screenshots)

If your PR contains in-game changes you must provide screenshots/videos of the changes.
https://www.youtube.com/watch?v=EGogCKKu4VY

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [ ] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
Adds the new MT-7 Energy Magnum to the game. Slightly better then the XO1

:cl: Central Security Division(Insaniam)
- add: Added the MT-7 Energy magnum soon made to apear on your stations warden or detectives!

ScarletLightweaverSS14 committed:
Inital

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
threw around edits

ScarletLightweaverSS14 committed:
Merge branches 'EnergyMagnum' and 'EnergyMagnum' of https://github.com/ScarletLightweaverSS14/Starlight-Chimes into EnergyMagnum

Clone0401 commented:
I wana mention for those looking at this who didn’t follow the related discord suggestions post.
One of the things that was brought up as a possibility for how to implement this is to repurpose the theives toolbox code to allow the hos to select what special weapon they want to use.

And regarding what you said about gun replacements. Honestly I don’t think there’s another way caps laser should be replaced, imo it’s a fairly unique gun already.

ScarletLightweaverSS14 committed:
Balanciing

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Merge branch 'EnergyMagnum' of https://github.com/ScarletLightweaverSS14/Starlight-Chimes into EnergyMagnum

ScarletLightweaverSS14 committed:
Update battery_guns.yml

Fixxed a minor spelling mistake :pain

ScarletLightweaverSS14 committed:
Fix on spelling mistakes

AGAIN fixxing these spelling mistakes this time the mistake was SoundGunShot

it was meant to be soundGunShot

ScarletLightweaverSS14 committed:
I keep finding spelling mistakes

I keep finding spelling mistakes IM SO SSORRY

Conflee commented:
Sprites approved. Could use a smallll shading touch up later but good for now.

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 commented:
I forgot to change one last thing in the code will do now

ScarletLightweaverSS14 committed:
Change of fire rate

When balancing i decided 1.6 is a fine line

forgot to put it in til now srry

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Put into use one of the unused energy dome sprites

ScarletLightweaverSS14 committed:
Delete Resources/Prototypes/Entities/Effects/dome.yml

this was accidentaly added to the PR MB

ScarletLightweaverSS14 commented:
@Happyrobot33 hey uh, YAML linter fail because it cant find the file i deleted from the PR so i think it bugged it self out

JustAdler commented on Resources/Prototypes/Entities/Effects/dome.yml:
you deleted the dome

ScarletLightweaverSS14 replied:
Yeah because i never meant to add that change. It was for a other PR

ScarletLightweaverSS14 replied:
did i accidentaly like... delete the file from starlight or what.

if so is there a way i can re add it or whatever

JustAdler commented on Resources/Prototypes/Entities/Effects/dome.yml:
```diff
@@ -1,142 +0,0 @@
-      color: "#64b9de"
-  - type: PointLight
-    enabled: true
-    radius: 5
-    power: 10
-    color: "#64b9de"
-
-- type: entity
-  id: EnergyDomeSlowing
-  categories: [ HideSpawnMenu ]
-  parent: EnergyDomeBase
-  components:
-  - type: Fixtures
-    fixtures:
-      fix1:
-        shape:
-          !type:PhysShapeCircle
-          radius: 2.8
-        density: 0
-        hard: false
-        mask:
-        - None
-        layer:
-        - MidImpassable
-  - type: Sprite
-    drawdepth: LowFloors
-    sprite: Effects/EnergyDome/energydome_slowdown_big.rsi
-    layers:
-    - state: big
-      color: "#a3d177"
-  - type: PointLight
-    enabled: true
-    radius: 5
-    power: 30
-    color: "#a3d177"
-  - type: DamageContacts
-    damage:
-      types:
-        Slash: -1.5
-        Piercing: -1.5
```
```suggestion
- type: entity
  id: EnergyDomeBase
  abstract: true
  categories: [ HideSpawnMenu ]
  components:
  - type: Sprite
    drawdepth: Effects
    noRot: true
  - type: Fixtures
    fixtures:
      fix1:
        shape:
          !type:PhysShapeCircle
          radius: 0.8
        density: 0
        mask:
        - None
        layer:
        - BulletImpassable
        - Opaque
  - type: Physics
    bodyType: Static
  - type: Damageable
    damageContainer: Inorganic
    damageModifierSet: HardLightBarrier
  - type: AmbientSound
    volume: 35
    range: 5
    sound:
      path: /Audio/Machines/energyshield_ambient.ogg
  - type: EnergyDome
  - type: Tag
    tags:
      - HideContextMenu
      - IgnoreMelee

- type: entity
  id: EnergyDomeSmallPink
  categories: [ HideSpawnMenu ]
  parent: EnergyDomeBase
  components:
  - type: Sprite
    sprite: Effects/EnergyDome/energydome_small.rsi
    layers:
    - state: small
      color: "#f5166b"
  - type: PointLight
    enabled: true
    radius: 5
    power: 2
    color: "#f5166b"

- type: entity
  id: EnergyDomeSmallCap
  categories: [ HideSpawnMenu ]
  parent: EnergyDomeBase
  components:
  - type: Sprite
    sprite: Effects/EnergyDome/energydome_small.rsi
    layers:
    - state: small
      color: "#FFFF1E"

- type: entity
  id: EnergyDomeSmallRed
  categories: [ HideSpawnMenu ]
  parent: EnergyDomeBase
  components:
  - type: Sprite
    drawdepth: Effects
    noRot: true
    sprite: Effects/EnergyDome/energydome_small.rsi
    layers:
    - state: small
      color: "#b00000"
  - type: PointLight
    enabled: true
    radius: 5
    power: 2
    color: "#b00000"

- type: entity
  id: EnergyDomeMediumBlue
  categories: [ HideSpawnMenu ]
  parent: EnergyDomeBase
  components:
  - type: Fixtures
    fixtures:
      fix1:
        shape:
          !type:PhysShapeCircle
          radius: 1.8
        density: 0
        mask:
        - None
        layer:
        - BulletImpassable
        - Opaque
  - type: Sprite
    sprite: Effects/EnergyDome/energydome_medium.rsi
    layers:
    - state: medium
      color: "#64b9de"
  - type: PointLight
    enabled: true
    radius: 5
    power: 10
    color: "#64b9de"

- type: entity
  id: EnergyDomeSlowing
  categories: [ HideSpawnMenu ]
  parent: EnergyDomeBase
  components:
  - type: Fixtures
    fixtures:
      fix1:
        shape:
          !type:PhysShapeCircle
          radius: 2.8
        density: 0
        hard: false
        mask:
        - None
        layer:
        - MidImpassable
  - type: Sprite
    drawdepth: LowFloors
    sprite: Effects/EnergyDome/energydome_slowdown_big.rsi
    layers:
    - state: big
      color: "#a3d177"
  - type: PointLight
    enabled: true
    radius: 5
    power: 30
    color: "#a3d177"
  - type: DamageContacts
    damage:
      types:
        Slash: -1.5
        Piercing: -1.5
        
```

ScarletLightweaverSS14 committed:
Revert "Delete Resources/Prototypes/Entities/Effects/dome.yml"

This reverts commit 55064e35438693753ed9aa7583c66bc3851eacd3.

ScarletLightweaverSS14 commented:
how would happy feel about seeing this man

ScarletLightweaverSS14 committed:
Removed accidental dome.yml commit

I removed the accidental commit from dome.yml because it was meant for a other PR that is already open.

ScarletLightweaverSS14 committed:
Final dome.yml Fix sorry- im still new to it all

Once again trying to fix the dome.yml :P

im very sorry if this seems like commit slop im still very new to all this and im trying my best

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 commented:
alot of branch updates

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 commented:
okay so a friend told me that You’re supposed to update your branch via the sync branch button at the main directory page

and uh only do it when needed

i am so sorry if this looks so shit i really am im just trying my best to keep the pr up to date i dont wanna fuck somthing up @Happyrobot33

stickycheZ101 commented:
It’s all good mate, we all learned/are learning somehow

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ScarletLightweaverSS14 commented:
i fucking merged into the wrong branch PAIN

ScarletLightweaverSS14 committed:
Changed copyright to include myself for making the energymagnum

Changed copyright to include myself for making the energymagnum

ScarletLightweaverSS14 commented:
bump

StarlightHost committed:
Merge branch 'starlight-dev' into Starlight

ScarletLightweaverSS14 committed:
Merge branch 'Starlight' into EnergyMagnum

ShadesMars reviewed (approved):
Admin only, this looks fine.
